qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/template/TemplateFileTextDocumentService.java

```diff
@@ -119,38 +124,40 @@ public void didSave(DidSaveTextDocumentParams params) {
 
 	@Override
 	public CompletableFuture<Either<List<CompletionItem>, CompletionList>> completion(CompletionParams params) {
-		return computeModelAsync2(getDocument(params.getTextDocument().getUri()).getModel(),
-				(cancelChecker, template) -> {
+		return getTemplateCompose(params.getTextDocument(),
+				(template, cancelChecker) -> {
 					return getQuteLanguageService()
 							.doComplete(template, params.getPosition(), sharedSettings.getCompletionSettings(),
 									sharedSettings.getFormattingSettings(), sharedSettings.getNativeSettings(),
 									cancelChecker) //
-							.thenApply(list -> {
+							.thenApplyAsync(list -> {

```

Comment:
Why `thenApplyAsync` ?

Reply:
To be honest, I didn't understand the difference between `thenApply` and `thenApplyAsync`. I learned what the difference was, and I changed the code here to use `thenApply`.

Reply:
thenApplyAsync will create a new Thread, I'm not sure that we need that.